smpclient/__init__.py

```diff
@@ -50,10 +56,18 @@ async def upload(
     ) -> AsyncIterator[int]:
         """Iteratively upload an `image` to `slot`, yielding the offset."""
 
+        max_packet_size = self._transport.max_unencoded_size
+        if isinstance(self._transport, SMPBLETransport):
+            mcumgr_parameters = await asyncio.wait_for(self.request(MCUMgrParametersRead()), timeout=1.0)
+            if success(mcumgr_parameters):
+                max_packet_size = mcumgr_parameters.buf_size
+
+        # why 23? should be max_packet_size
         if self._transport.max_unencoded_size < 23:
             raise Exception("Upload requires an MTU >=23.")
 
-        response = await self.request(
+        #  Timeout for the initial chunk is long, as the device may need to erase the flash.
+        response = await asyncio.wait_for(self.request(
```

Comment:
I was not able to use FLASH_ERASE_PROGRESSIVELY at all on Nrf-sdk

Reply:
I did not state the correct Kconfig, try:

```
config IMG_ERASE_PROGRESSIVELY
	bool "Erase flash progressively when receiving new firmware"
	depends on MCUBOOT_IMG_MANAGER
	select STREAM_FLASH_ERASE
	help
	  If enabled, flash is erased as necessary when receiving new firmware,
	  instead of erasing the whole image slot at once. This is necessary
	  on some hardware that has long erase times, to prevent long wait
	  times at the beginning of the DFU process.
```

MCUBoot may be different.  I wonder if there is a desire to unify the SMP servers between MCUBoot and Zephyr app?

```
config BOOT_ERASE_PROGRESSIVELY
	bool "Erase flash progressively when receiving new firmware"
	default y if SOC_FAMILY_NRF
	help
	 If enabled, flash is erased as necessary when receiving new firmware,
	 instead of erasing the whole image slot at once. This is necessary
	 on some hardware that has long erase times, to prevent long wait
	 times at the beginning of the DFU process.
```